### PR TITLE
search user with name

### PR DESCRIPTION
**Pull request by ghost**
## Description
Search user with name
## Reference Requirements
- Task ID: #17 

## Impacted Areas in Application
List general components of the application that this PR will affect:

- PR: #11 

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- EXPLAIN select count(*) as aggregate from `users` where `name` LIKE '%admin%'\G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: users
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: 31
     filtered: 11.11
        Extra: Using where
1 row in set, 1 warning (0.00 sec)
- EXPLAIN select * from `users` where `name` LIKE '%admin%' limit 10 offset 0\G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: users
   partitions: NULL
         type: ALL
possible_keys: NULL
          key: NULL
      key_len: NULL
          ref: NULL
         rows: 31
     filtered: 11.11
        Extra: Using where
1 row in set, 1 warning (0.00 sec)
- EXPLAIN select * from `user_infos` where `user_infos`.`user_id` in ('1')\G;
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: user_infos
   partitions: NULL
         type: ref
possible_keys: user_infos_user_id_foreign
          key: user_infos_user_id_foreign
      key_len: 4
          ref: const
         rows: 1
     filtered: 100.00
        Extra: NULL
1 row in set, 1 warning (0.00 sec)

### CODE REVIEW

File: app/Http/Controllers/Admin/UsersController.php [hunk 1]
```diff
@@ -12,11 +12,14 @@ class UsersController extends Controller
...
+     * @param \Illuminate\Http\Request $request request
+     *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-        $users = User::with('userInfo')->paginate(config('define.number_pages'));
+        $nameUser = $request->input('name_user_search');
+        $users = User::with('userInfo')->where('name', 'LIKE', '%' . $nameUser . '%')->paginate(config('define.number_pages'));
```
Comment (at-tuanphan) [inline app/Http/Controllers/Admin/UsersController.php]: theo anh nghĩ tụi em nên viết một Trait class để phục vụ cho việc search thì hay hơn, vì mình có nhiều page và nhiều page có nhưng cách search khác nhau.
tham khảo: https://gist.github.com/mul14/726c4d2514c5d9f28b43
library: https://github.com/nicolaslopezj/searchable

**Review by ghost — COMMENTED** [review]
please check my comment

File: app/Http/Controllers/Admin/UsersController.php [hunk 2]
```diff
@@ -15,11 +15,14 @@ class UsersController extends Controller
...
      *
+     * @param \Illuminate\Http\Request $request request
+     *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-        $users = User::with('userInfo')->paginate(config('define.number_pages'));
+        $nameUser = $request->input('name_user_search');
```
Comment (at-tuanphan) [inline app/Http/Controllers/Admin/UsersController.php]: $request->input('name_user_search'); => $request->name_user_search;

**Review by ghost — COMMENTED** [review]
please check my comment

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**